sct-commons/src/test/java/org/lfenergy/compas/sct/commons/dto/LDEPFSettingsSupplierTest.java

```diff
@@ -0,0 +1,353 @@
+                //required attributes to match LDEVICE source for ExtRef
+                .ldInst("INST_OF_LDEVICE_SOURCE")
+                .lnClass("PTRC").lnInst("0").lnPrefix(null)
+                .doName("Str").doInst("0")
+                .daName("general")
+                .build();
+        TCompasBay compasBay = setCompasBayUUID(sclRootAdapter, "IED_NAME1", "bayUUID1");
+        // When
+        LDEPFSettingsSupplier settingsSupplier = () -> List.of(digitalSetting);
+        var iedSources = settingsSupplier.getIedSources(sclRootAdapter, compasBay, digitalSetting);
+        // Then
+        assertThat(iedSources).isEmpty();
+    }
+
+    @Test
+    void getIedSources_should_return_NoIedSource_whenLDeviceSourceIsOff() {
+        // Given
+        SCL scd = SclTestMarshaller.getSCLFromFile("/scd-ldepf/scd_ldepf_iedSources_in_different_bay.xml");
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
+        LDEPFSetting digitalSetting = LDEPFSetting.builder()
+                .channelDigitalNum("10")
+                .channelAnalogNum(null)
+                .rteIedType("BCU").iedRedundancy("None").iedInstance(BigInteger.valueOf(1))
+                //required attributes to match LDEVICE source for ExtRef
+                .ldInst("LDPXINVALID")
+                .lnClass("PTRC").lnInst("0").lnPrefix(null)
+                .doName("Str").doInst("0")
+                .daName("general")
+                .build();
+        TCompasBay compasBay = setCompasBayUUID(sclRootAdapter, "IED_NAME1", "bayUUID1");
+        // When
+        LDEPFSettingsSupplier settingsSupplier = () -> List.of(digitalSetting);
+        var iedSources = settingsSupplier.getIedSources(sclRootAdapter, compasBay, digitalSetting);
+        // Then
+        assertThat(iedSources).isEmpty();
+    }
+
+    @Test
+    void getIedSources_should_return_NoIedSource_whenNoExistingLNodeSource() {
+        // Given
+        SCL scd = SclTestMarshaller.getSCLFromFile("/scd-ldepf/scd_ldepf_iedSources_in_different_bay.xml");
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
+        var digitalSetting = LDEPFSetting.builder()
+                .channelDigitalNum("10")
+                .channelAnalogNum(null)
+                .rteIedType("BCU").iedRedundancy("None").iedInstance(BigInteger.valueOf(1))
+                .ldInst("LDPX")
+                //required attributes to verify LNODE source for ExtRef
+                .lnClass("PTRC").lnInst("100000").lnPrefix(null)
+                .doName("Str").doInst("0")
+                .daName("general")
+                .build();
+        TCompasBay compasBay = setCompasBayUUID(sclRootAdapter, "IED_NAME1", "buyUUID1");
+        // When
+        LDEPFSettingsSupplier settingsSupplier = () -> List.of(digitalSetting);
+        var iedSources = settingsSupplier.getIedSources(sclRootAdapter, compasBay, digitalSetting);
+        // Then
+        assertThat(iedSources).isEmpty();
+    }
```

Comment:
Why not use a parameterized test for these 3 tests having same assert?

Reply:
The only ones grouped were those that concern LDevice Source: `_whenNoExistingLDeviceSource` and `_whenLDeviceSourceIsOff`